os/arch/arm/src/amebasmart/amebasmart_pmhelpers.c

```diff
@@ -115,19 +115,29 @@ int SOCPS_AONWakeReason(void)
 
 void pg_timer_int_handler(void *Data)
 {
-	pmvdbg("Timer wakeup interrupt handler!!\n");
+	pmvdbg("PM Timer interrupt handler!!\n");
+	switch (g_pm_timer.use_timer) {
```

Comment:
For multiple timer case, we need to identify which source trigger the callback, thus we need to provide the callback data for the last argument, which indicate the trigger type.

Reply:
Agree to your point
For multiple timers coming in concurrently, this and some other considerations need to be handled.
However, I suggest let's not handle those cases right now since we don't have the clear requirement. It could again become a guessing game.